File: sample/src/main/java/com/example/rwellnitz/tasktest/view_model/Advertisement.java
package com.example.rwellnitz.tasktest.view_model;

import com.example.rwellnitz.tasktest.auto.MyViewHolder;

import com.renard.auto_adapter.AdapterItem;
import com.renard.auto_adapter.Unique;

@AdapterItem(value = "NewsArticleAdapter", viewHolder = MyViewHolder.class)
public class Advertisement implements Unique {
    private final long id;

    public String message = "This is an advertisement message";

    public Advertisement(final long id) {
        this.id = id;
    }

    @Override
    public long getId() {
        return this.id;
    }
}

File: sample/src/main/java/com/renard/auto_adapter/sample/Advertisement.kt
package com.renard.auto_adapter.sample

import com.renard.auto_adapter.AdapterItem
import com.renard.auto_adapter.Unique

@AdapterItem("NewsArticleAdapter")
class Advertisement(private val id: Long) : Unique {

    var message = "This is an advertisement message"

    override fun getId(): Long = this.id
}
